### PR TITLE
Remove Cli

### PR DESCRIPTION
**Pull request by shanaxel42**
Remove CLI code and pipeline component class.
Rename package names so that the algorithms can be used in the same way, ex.
```
from starifish import Filter
wth = Filter.WhiteTopHat()
``` 
Changed AlgorithmBase to be the MetaClass for logging. 

Everything api-wise should work the same. Everything cli-wise should be removed. 

One exception was the starfish validate command. Also I kept the ascii art cause that's essential

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -57,9 +57,6 @@ jobs:
     - name: 3D smFISH data processing example
       if: type = push and branch =~ /^(master|merge)/
       script: make install-dev 3d_smFISH.py
-    - name: iss_cli.sh data processing example
```
Comment (ttung) [inline .travis.yml]: You can plausibly remove the Makefile rules for running `.sh` data processing examples.

File: docs/source/index.rst [hunk 2]
```diff
@@ -30,7 +30,7 @@ see the :ref:`Getting Started <getting started>` section.
 
 Starfish's documentation also contains instructions on :ref:`creating image-based transcriptomics
 processing pipelines <creating_an_image_processing_pipeline>`, tutorials on running
-starfish using the CLI, API, and workflow runner, and a worked example of contributing code to
+starfish using the API, and workflow runner, and a worked example of contributing code to
```
Comment (ttung) [inline docs/source/index.rst]:
```suggestion
starfish using the API and workflow runner, and a worked example of contributing code to
```

File: starfish/core/image/Filter/test/test_api_contract.py [hunk 3]
```diff
@@ -24,10 +24,26 @@
...
 from starfish.core.image import Filter
-from starfish.core.image._filter.max_proj import MaxProject
-from starfish.core.image._filter.reduce import Reduce
-
-methods: Mapping[str, Type] = Filter._algorithm_to_class_map()
+from starfish.core.image.Filter.max_proj import MaxProject
+from starfish.core.image.Filter.reduce import Reduce
+
+
+methods: Mapping[str, Type] = {
```
Comment (ttung) [inline starfish/core/image/Filter/test/test_api_contract.py]: Can you just do a `dir(Filter)` to get this?

File: docs/source/creating_an_image_processing_pipeline/index.rst [hunk 4]
```diff
@@ -75,20 +75,6 @@ origin. At this point, it's trivial to create a cell x gene matrix.
 
     tutorials/exec_feature_identification_and_annotation.rst
 
-Putting Together a Pipeline Recipe and Running it
```
Comment (ttung) [inline docs/source/creating_an_image_processing_pipeline/index.rst]: Are we dropping the recipe runner?

File: docs/source/usage/validation/index.rst [hunk 5]
```diff
@@ -17,47 +17,11 @@ starfish validate --help will provide instructions on how to use the tool:
...
     starfish validate experiment tmp/experiment.json > /dev/null && echo ok
 
 
-Building a :ref:`synthetic experiment <cli_build>` can provide you with a template that
-you can use to model your own data. If you then modify that experiment incorrectly, you
-might see the following validation warning:
-
-::
-
-    $ starfish validate experiment tmp/experiment.json
```
Comment (ttung) [inline docs/source/usage/validation/index.rst]: Do we want to preserve some documentation about `starfish validate`?

File: docs/source/creating_an_image_processing_pipeline/index.rst (hunk 4 above)
Comment (shanaxel42) [reply docs/source/creating_an_image_processing_pipeline/index.rst]: I took out all the recipe stuff :/ should leave the runner in?

File: docs/source/usage/validation/index.rst (hunk 5 above)
Comment (shanaxel42) [reply docs/source/usage/validation/index.rst]: I still have just the validate documentation in but I took out the builder bit

File: .travis.yml (hunk 1 above)
Comment (ttung) [reply .travis.yml]: ping

File: docs/source/index.rst [hunk 6]
```diff
@@ -30,7 +30,7 @@ see the :ref:`Getting Started <getting started>` section.
 
 Starfish's documentation also contains instructions on :ref:`creating image-based transcriptomics
 processing pipelines <creating_an_image_processing_pipeline>`, tutorials on running
-starfish using the CLI, API, and workflow runner, and a worked example of contributing code to
+starfish using the API and workflow runner, and a worked example of contributing code to
```
Comment (ttung) [inline docs/source/index.rst]: If we are removing the recipes code, then the workflow runner blurb is inaccurate.

```suggestion
starfish using the API, and a worked example of contributing code to
```

File: docs/source/usage/validation/index.rst (hunk 5 above)
Comment (ttung) [reply docs/source/usage/validation/index.rst]: It is plausible that we want to retain the example about validating an existing experiment.

File: starfish/core/image/Filter/test/test_api_contract.py (hunk 3 above)
Comment (ttung) [reply starfish/core/image/Filter/test/test_api_contract.py]: ping

File: .travis.yml (hunk 1 above)
Comment (shanaxel42) [reply .travis.yml]: removed!

File: starfish/core/image/Filter/test/test_api_contract.py (hunk 3 above)
Comment (shanaxel42) [reply starfish/core/image/Filter/test/test_api_contract.py]: I tried this and it doesn't quite give me what I need. ```['Bandpass', 'Clip', 'ClipPercentileToZero', 'ClipValueToZero', 'DeconvolvePSF', 'ElementWiseMultiply', 'GaussianHighPass', 'GaussianLowPass', 'Laplace', 'LinearUnmixing', 'MatchHistograms', 'MaxProject', 'MeanHighPass', 'Reduce', 'WhiteTophat', 'ZeroByChannelMagnitude', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '_base', 'bandpass', 'clip', 'clip_percentile_to_zero', 'clip_value_to_zero', 'element_wise_mult', 'gaussian_high_pass', 'gaussian_low_pass', 'laplace', 'linear_unmixing', 'match_histograms', 'max_proj', 'mean_high_pass', 'reduce', 'richardson_lucy_deconvolution', 'util', 'white_tophat', 'zero_by_channel_magnitude']``` but I think the refactor to make another clever auto import method will work when I do that!

**Comment by ttung** [comment]
This broke the docker test.
https://travis-ci.com/spacetx/starfish/jobs/217075503

**Pull request merged**